File: API/Python/examples/rpc/rpc.py
#encoding=utf8
import sys 
sys.path.append('../../') 

from zbus import SingleBroker, Rpc 

broker = SingleBroker(host='127.0.0.1', port=15555)

rpc = Rpc(broker=broker, 
          mq='MyRpc', 
          module='Interface',
          encoding='utf8',
          timeout=10)


print rpc.getString('hello')

broker.destroy()

File: api/csharp/examples/RPC/Rpc.cs
using Zbus.Broker;
using Zbus.RPC;
using System;
namespace Zbus.Examples.RPC
{ 
    class RpcClientTest
    { 
        public static void Main(string[] args)
        {  
            IBroker broker = new SingleBroker();
            Rpc rpc = new Rpc(broker); 

            Request req = new Request();  
            req.Mq = "MyRpc";
            req.Module = "MyService"; 
            req.Method = "getString";
            req.Args = new object[] { "test" };

            for(int i = 0; i < 100; i++)
            {
               object res = rpc.Invoke(req);
               Console.WriteLine(res);

            } 
           
            broker.Dispose(); 
            Console.ReadKey();
        }
    }
}
